library/pod-security-policy/fsgroup/src.rego

```diff
@@ -0,0 +1,47 @@
+package k8spspfsgroup
+
+violation[{"msg": msg, "details": {}}] {
+    spec := input.review.object.spec
+    not input_fsGroup_allowed(spec)
+    msg := sprintf("The provided pod spec fsGroup is not allowed, pod: %v. Allowed fsGroup: %v", [input.review.object.metadata.name, input.parameters])
+}
+
+input_fsGroup_allowed(spec) {
+    # RunAsAny - No range is required. Allows any fsGroup ID to be specified.
+    input.parameters.rule == "RunAsAny"
+}
+input_fsGroup_allowed(spec) {
+    # MustRunAs - Validates pod spec fsgroup against all ranges
+    input.parameters.rule == "MustRunAs"
+    fg := spec.securityContext.fsGroup
+    count(input.parameters.ranges) > 0
+    range := input.parameters.ranges[_]
+    value_within_range(range, fg)
+}
+input_fsGroup_allowed(spec) {
+    # MayRunAs - Validates pod spec fsgroup against all ranges or allow pod spec fsgroup to be left unset
+    input.parameters.rule == "MayRunAs"
+    not has_field(spec, "securityContext")
+}
+input_fsGroup_allowed(spec) {
+    # MayRunAs - Validates pod spec fsgroup against all ranges or allow pod spec fsgroup to be left unset
+    input.parameters.rule == "MayRunAs"
+    has_field(spec, "securityContext")
+    not has_field(spec.securityContext, "fsGroup")
+}
```

Comment:
Here you could combine the 2 rules above and use 1 check that the fsGroup field (or any securityContext) is missing:

not spec.securityContext.fsGroup